pkg/syncer/storagepool/storageclasswatch.go

```diff
@@ -325,7 +327,12 @@ func (w *StorageClassWatch) getDatastoreToPolicyCompatibility(ctx context.Contex
 	for _, policyID := range w.policyIds {
 		compat, err := w.vc.PbmCheckCompatibility(ctx, datastoreMorList, policyID)
 		if err != nil {
-			log.Errorf("Failed to check PBM compatibility: %s %s", datastoreMorList, policyID)
+			if soap.IsSoapFault(err) && strings.Contains(err.Error(), "Failed to fetch content for the provided profiles.") {
+				// stale policyIDs can be skipped safely
+				log.Infof("Skipping policy %s that failed in check PBM compatibility %s with error %v",
```

Comment:
can you use `%v` format specifier for `datastoreMorList`? 
`%s` is for printing string.

Reply:
done.